lightningd/invoice.c

```diff
@@ -180,6 +289,14 @@ static void json_invoice(struct command *cmd,
 		b11->description = tal_strndup(b11, buffer + desc->start,
 					       desc->end - desc->start);
 
+	/* fallback address */
+	if (fallback) {
+		bool testnet;
+		b11->fallback = scriptpubkey_from_address(b11, &testnet,
+							  buffer + fallback->start,
+							  fallback->end - fallback->start);
+	}
```

Comment:
Yes, and also if b11->fallback is NULL, we fail.

Truthfully, scriptpubkey_from_address should return a pointer to the bip173 network name, or NULL, and take a u8 ** for the script instead of returning it.  Then we could give a much more sensible message, and be able to distinguish litecoin from bitcoin and testnet from regtest, for example...